### PR TITLE
Remove ValidationException

### PR DESCRIPTION
**Pull request by stephentoub**
### Motivation and Context

Remove custom ValidationException in favor of using built-in .NET exception types typically used for the same purposes.

### Description

The conditions ValidationException was used for validating are better suited to existing .NET exception types that .NET developers expect in those situations.

- NotNullOrWhiteSpace is represented with ArgumentException.
- Directory not found is represented with DirectoryNotFoundException
- Invalid state of an object (e.g. a field not being initialized) is represented with InvalidOperationException

Argument validation exceptions are also usage exceptions typically not intended to be caught.  The same would go for the ValidationExceptions that were serving the same purpose, so having a separate type identity for those exceptions isn't desirable.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

cc: @shawncal, @dluc

### CODE REVIEW

**Review by adrianwyatt — DISMISSED** [review]

File: dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs [hunk 1]
```diff
@@ -53,28 +55,29 @@ internal static void ValidFunctionName([NotNull] string? functionName)
...
+            throw new ArgumentException(textParamName, message);
         }
     }
 
     internal static void DirectoryExists(string path)
     {
         if (!Directory.Exists(path))
         {
-            ThrowValidationException(ValidationException.ErrorCodes.DirectoryNotFound, $"Directory not found: {path}");
+            throw new DirectoryNotFoundException($"Directory '{path}' could not be found.");
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs:80]: nit: I think we could just pass "path" to DirectoryNotFoundException and do the format in the exception class

Comment (stephentoub) [reply dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs:80]: DirectoryNotFoundException is a built-in .NET exception.  Its ctor just takes a message; it doesn't have one that takes a path and wraps that in a message.

File: dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs [hunk 2]
```diff
@@ -24,18 +26,18 @@ internal static void NotNull([NotNull] object? obj, [CallerArgumentExpression(na
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal static void NotEmpty([NotNull] string? str, string? message = null, [CallerArgumentExpression(nameof(str))] string? paramName = null)
+    internal static void NotNullOrWhiteSpace([NotNull] string? str, [CallerArgumentExpression(nameof(str))] string? paramName = null)
```
Comment (shawncal) [inline dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs:29]: OMG thank you! This has always been annoying.

File: dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs (hunk 1 above)
Comment (shawncal) [reply dotnet/src/SemanticKernel.Abstractions/Diagnostics/Verify.cs:80]: ah, gotcha!

**Review by shawncal — APPROVED** [review]

**Pull request merged**